runtime/balrt.h

```diff
@@ -496,51 +494,35 @@ static READONLY inline double taggedToFloat(TaggedPtr p) {
     return *np;
 }
 
-static READONLY inline int64_t taggedFloatComparator(TaggedPtr lhs, TaggedPtr rhs) {
+static READONLY inline CompareResult taggedFloatComparator(TaggedPtr lhs, TaggedPtr rhs) {
     double lhsVal = taggedToFloat(lhs);
     double rhsVal = taggedToFloat(rhs);
-    if (lhsVal == rhsVal) {
-        return COMPARE_EQ;
-    }
-    if (lhsVal < rhsVal) {
-        return COMPARE_LT;
-    }
-    if (lhsVal > rhsVal) {
-        return COMPARE_GT;
-    }
-    return COMPARE_UN;
+    return COMPARE(lhsVal, rhsVal);
```

Comment:
This doesn't get COMPARE_UN right. double and int are not the same because int is totally ordered and double isn't (because of NaN).  Better to call COMPARE something like COMPARE_TOTAL to avoid this confusion.

Reply:
fixed